src/main/java/com/amazonaws/neptune/auth/NeptuneSigV4SignerBase.java

```diff
@@ -341,5 +355,12 @@ public String getXAmzDateHeader() {
         public String getAuthorizationHeader() {
             return authorizationHeader;
         }
+
+        /**
+         * @return the Session Token value
+         */
+        public String getSessionToken() {
```

Comment:
Retune Optional<String>, it should be Optional.empty() if the session token is null or blank. Optional.of(sessionToken) otherwise.

Reply:
Skipped optional.  I think it is just overusing of Optional. We already check for credential type (BasicSessionCredentials). And if is BasicSessionCredentials type and sessionToken is null then we should get an error. https://stackoverflow.com/questions/26327957/should-java-8-getters-return-optional-type

Reply:
Sounds good. Let's go with this.